lib/plugins/wallet/cardano/cardano.js

```diff
@@ -0,0 +1,91 @@
+const { walletId } = require('./wallet')
+const cardanoClient = require('./client')
+
+const MINIMUM_CONFIRMATIONS_FOR_APPROVE = 1
+
+// TODO: work with BN here instead of api.js?
+
+function assertCryptoCode(cryptoCode) {
+  if (cryptoCode !== 'ADA') {
+    throw new Error(`Unsupported crypto: ${cryptoCode}`)
+  }
+}
+
+/**
+ * We don't use `account` parameter, because there is any interesting information for us,
+ * only global key is `seed`, but this seed is created from 24-word mnemonic, so it
+ * isn't usefull - instead of this we use walletId() function to obtain wallet credentials
+ */
+async function balance(account, cryptoCode) {
+  assertCryptoCode(cryptoCode)
+
+  return walletId().then(({ walletId, index }) =>
+    cardanoClient.walletBalance(walletId, index, MINIMUM_CONFIRMATIONS_FOR_APPROVE)
+  ).then(result => result.confirmed)
+}
+
+async function sendCoins(account, address, cryptoAtoms, cryptoCode) {
+  assertCryptoCode(cryptoCode)
+
+  return walletId().then(({ walletId, index }) =>
+    cardanoClient.sendCoins(walletId, index, address, cryptoAtoms)
+  )
+}
+
+async function newAddress(account, info) {
+  assertCryptoCode(cryptoCode)
+
+  return walletId().then(({ walletId, index }) =>
+    cardanoClient.newAddress(walletId, index)
+  )
+}
+
+async function getStatus(account, toAddress, requested, cryptoCode) {
+  assertCryptoCode(cryptoCode)
+
+  // we include only income transactions, because we need only about
+  // currency that customer sent to this address, outcome transactions
+  // are not interesting to us
+  return cardanoClient
+    .addressBalance(
+      toAddress,
+      MINIMUM_CONFIRMATIONS_FOR_APPROVE,
+      cardanoClient.INCLUDE_INCOME_TRANSACTIONS
+    )
+    .then(balance => {
+      if (balance.confirmed.gte(requested)) {
+        return { status: 'confirmed' }
+      } else if (balance.total.gte(request)) {
+        return { status: 'authorized' }
+      } else if (balance.confirmed.eq(balance.total)) {
```

Comment:
what is the difference between `insufficientFunds` and `notSeen` ? Importantly, what should we show with  `balance.confirmed == balance.total == 0`?

Reply:
insufficientFunds is for total > 0 (include pending), notSeen for 0, will update that